src/main/scala/com/codacy/parsers/implementation/CoberturaParser.scala

```diff
@@ -3,41 +3,34 @@ package com.codacy.parsers.implementation
 import java.io.File
 
 import com.codacy.api.{CoverageFileReport, CoverageReport}
-import com.codacy.parsers.CoverageParser
-import com.codacy.parsers.util.{TextUtils, XMLoader}
+import com.codacy.parsers.util.TextUtils
+import com.codacy.parsers.{CoverageParser, XmlReportParser}
 
-import scala.util.{Failure, Success, Try}
 import scala.xml.{Elem, NodeSeq}
 
-object CoberturaParser extends CoverageParser {
+object CoberturaParser extends CoverageParser with XmlReportParser {
 
   override val name: String = "Cobertura"
 
-  def parse(projectRoot: File, reportFile: File): Either[String, CoverageReport] = {
-    val report = (Try(XMLoader.loadFile(reportFile)) match {
-      case Success(xml) if hasCorrectSchema(xml) =>
-        Right(xml \\ "coverage")
-
-      case Success(_) =>
-        Left("Invalid report. Could not find top level <coverage> tag.")
+  private val CoverageTag = "coverage"
+  private val LineRateAttribute = "@line-rate"
 
-      case Failure(ex) =>
-        Left(s"Unparseable report. ${ex.getMessage}")
-    })
-
-    report.flatMap(parse(projectRoot, _))
+  def parse(projectRoot: File, reportFile: File): Either[String, CoverageReport] = {
+    parseReport(reportFile, s"Could not find top level <$CoverageTag> tag") {
+      parse(projectRoot, _)
+    }
   }
 
   // restricting the schema to <coverage line-rate=...>
   // ensures this will not consider Clover reports which also have a <coverage> tag
-  private def hasCorrectSchema(xml: Elem) = {
-    (xml \\ "coverage" \ "@line-rate").nonEmpty
-  }
+  override def validateSchema(xml: Elem): Boolean = (xml \\ CoverageTag \ LineRateAttribute).nonEmpty
 
-  private def parse(projectRoot: File, report: NodeSeq): Either[String, CoverageReport] = {
+  override def getRootNode(xml: Elem): NodeSeq = xml \\ CoverageTag
+
+  private def parse(projectRoot: File, report: NodeSeq) = {
     val projectRootStr: String = TextUtils.sanitiseFilename(projectRoot.getAbsolutePath)
 
-    val total = (TextUtils.asFloat((report \\ "coverage" \ "@line-rate").text) * 100).toInt
+    val total = math.round(TextUtils.asFloat((report \\ CoverageTag \ LineRateAttribute).text) * 100)
```

Comment:
If you change to LineRate you can use `\@` here.

Reply:
Done ✅